skbio/io/format/tests/test_stockholm.py

```diff
@@ -0,0 +1,148 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from six import assertRaisesRegex
+
+import unittest
+
+from skbio import TabularMSA
+from skbio.io import StockholmFormatError
+from skbio.io.format.stockholm import (_stockholm_to_tabular_msa,
+                                       _stockholm_sniffer)
+from skbio import Protein, DNA, RNA
+from skbio.util import get_data_path
+
+
+class TestStockholmSniffer(unittest.TestCase):
+    def setUp(self):
+        self.positives = [get_data_path(e) for e in [
+            'stockholm_extensive',
+            ]]
+
+        self.negatives = [get_data_path(e) for e in [
+            'stockholm_data_only',
+            'empty',
+            ]]
+
+    def test_positives(self):
+        for fp in self.positives:
+            self.assertEqual(_stockholm_sniffer(fp), (True, {}))
+
+    def test_negatives(self):
+        for fp in self.negatives:
+            self.assertEqual(_stockholm_sniffer(fp), (False, {}))
+
+
+class TestStockholmReader(unittest.TestCase):
```

Comment:
Can you add the following reader test cases?

- interleaved stockholm file (not currently supported): this should raise the same error as reading a file with duplicate sequence names. Example interleaved file [here](http://www.bioperl.org/wiki/Stockholm_multiple_alignment_format), but note that we can't use that data due to license incompatibilities, so just make up a small example in interleaved format.
- `stockholm_data_only`, `empty`, and `whitespace_only` files raise error stating that required header is missing
- file missing `//` raises an error
- file with blank and/or whitespace-only lines works correctly (those lines are ignored)
- file with unrecognized markup line raises an error (e.g., `#=GZ foo bar`)
- file with malformed metadata markup line raises an error (e.g., `#=GF foo \n`)
- file with malformed sequence data line raises an error (e.g., `my-seq \n`)
- file with sequences of differing length raises error
- file with multiple stockholm alignments only reads the first alignment
- file with GF/GC/GS/GR metadata in recommended placement vs. different placement results in the same `TabularMSA`
- file with GC or GR metadata of wrong length raises an error